bookmarklet/bookmarklet.ts

```diff
@@ -3,576 +3,688 @@ import './bookmarklet.css';
+    const idCss = 'com_kamilsarelo_dynatrace_timetracking_css';
+    const idHtml = 'com_kamilsarelo_dynatrace_timetracking_html'; // https://stackoverflow.com/questions/6028211/what-is-the-standard-naming-convention-for-html-css-ids-and-classes/37797488#37797488
+
+    // clear previous instances
+    function clear() {
+      // or:  const clear = function() {
+      [idHtml, idCss].forEach(function (id) {
+        document.querySelectorAll('#' + id).forEach(function (element) {
+          element.parentNode.removeChild(element); //https://stackoverflow.com/questions/8830839/javascript-dom-remove-element/8830882
+        });
+      });
+    }
+    clear();
+
+    (function () {
+      // or: function loadCss() { | const loadCss = function() {
+      return new Promise(function (resolve, reject) {
+        // https://stackoverflow.com/questions/574944/how-to-load-up-css-files-using-javascript/51183077#51183077
+
+        // CSS
+
+        const link = document.createElement('link');
+        link.id = idCss;
+        link.rel = 'stylesheet';
+        link.type = 'text/css';
+        // link.href = 'https://localhost/bookmarklet.css';
+        // link.href = 'https://cdn.jsdelivr.net/gh/kamilsarelo/timetracking/bookmarklet/bookmarklet.css';
+        link.href =
+          'https://christian-fischer.github.io/com.dynatrace.timetracking.bookmarklet.css?q=' +
+          new Date().getTime();
+        link.media = 'all';
+        link.onload = function () {
+          // https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onload and https://stackoverflow.com/questions/6348494/addeventlistener-vs-onclick/6348533#6348533
+          resolve();
+        };
+        document.head.appendChild(link); // https://developer.mozilla.org/de/docs/Web/API/Document/head
+      });
+    })() // anonymous function call: https://stackoverflow.com/questions/7498361/defining-and-calling-function-in-one-step
+      .then(function () {
+        // or: // loadCss().then(function () {
+
+        // HTML
+
+        const idContent = 'com_kamilsarelo_dynatrace_timetracking_content';
+        const idHeader = 'com_kamilsarelo_dynatrace_timetracking_header';
+        const idMain = 'com_kamilsarelo_dynatrace_timetracking_main';
+        const idInput = 'com_kamilsarelo_dynatrace_timetracking_input';
+        const idFooter = 'com_kamilsarelo_dynatrace_timetracking_footer';
+        const idFooterLeft =
+          'com_kamilsarelo_dynatrace_timetracking_footer_left';
+        const idFooterRight =
+          'com_kamilsarelo_dynatrace_timetracking_footer_right';
+        const idSelectType =
+          'com_kamilsarelo_dynatrace_timetracking_select_type';
+        const idButtonBook =
+          'com_kamilsarelo_dynatrace_timetracking_action_book';
+        const idButtonClear =
+          'com_kamilsarelo_dynatrace_timetracking_action_clear';
+        const idButtonClose =
+          'com_kamilsarelo_dynatrace_timetracking_action_close';
```

Comment:
this looks weird; tbh. line length 80 is so last century.
I propose to increase this to something more sensible like maybe 120

Reply:
I'd strongly advise again this! 80 cols is the Prettier default and they state explicitly on their website that it should not be altered (especially not to 120, 100 is already too much for the way Prettier formats code). Also TypeScript is not as verbose as Java is and we only use 2 spaces as indentation here. Vertical code is more readable to the human eye anyway. May be that in this particular case it looks rather ... awkward. But if the code would be more well-structured, the indentation would be much shorter here and Prettier wouldn't wrap it.